R/parameters.R

```diff
@@ -414,14 +414,43 @@ parameterise_human_equilibrium <- function(parameters, eq) {
 #' @description NOTE: the inital EIR is likely to change unless the rest of the
 #' model is in equilibrium
 #'
+#' max_total_M is calculated using the equilibrium solution from "Modelling the
+#' impact of vector control interventions on Anopheles gambiae population
+#' dynamics"
+#'
 #' @param parameters the parameters to modify
 #' @param EIR to work from
-#' @param limit_grace the number of mosquitos to allocate to the simulation as a
-#' proportion of total_M. e.g. 10 will allocate total_M * 10 mosquitos. High
-#' values of `limit_grace` are required for simulations with seasonality.
 #' @export
-parameterise_mosquito_equilibrium <- function(parameters, EIR, limit_grace=1.5) {
+parameterise_mosquito_equilibrium <- function(parameters, EIR) {
   parameters$total_M <- equilibrium_total_M(parameters, EIR)
-  parameters$mosquito_limit <- parameters$total_M * limit_grace
+  K0 <- calculate_carrying_capacity(parameters)
+  R_bar <- calculate_R_bar(parameters)
+  max_K <- max(vnapply(seq(365), function(t) {
+    carrying_capacity(
+      t,
+      parameters$model_seasonality,
+      parameters$days_per_timestep,
+      parameters$g0,
+      parameters$g,
+      parameters$h,
+      K0,
+      R_bar
+    )
+  }))
+  omega <- calculate_omega(parameters)
+  max_total_M <- max_K * (
+    1 / (
+      2 * parameters$dl * parameters$mum * (
+        1 + parameters$dpl * parameters$mup
+      )
+    )
+  ) * (
+    1 / (
+      parameters$gamma * (omega + 1)
+    )
+  ) * (
+    omega / (parameters$ml * parameters$del) - (1 / (parameters$ml * parameters$dl)) - 1
+  )
+  parameters$mosquito_limit <- ceiling(max_total_M * 5) #Allow for random fluctuations
```

Comment:
Is there some top limit here that shoudl throw an error? Maybe its checked elsewhere but if I put in a combination of wrong seasonality pars and and very high EIR will the run just choke my computer memory?

Reply:
That's a good question!

Seasonality and EIR combinations feel like a complicated parameter set to validate. We'd need to come up with pretty robust rules to only throw errors when the researcher is genuinely making a mistake and not plotting extremes or experimenting. My guess would be that we don't have those rules?

On large scale execution, the model will crash, get an R error, or be killed by the OS before it consumes all memory. I've just run a bunch of tests to confirm that.

On that note, we should make an issue for a maximum human/EIR to support. Let's say `human_population=1e6`, `EIR=1e5`?

Reply:
Agreed - I don't think we have those rules so as long as it fails (somewhat) elegantly I think thats fine!
Yes, those seem like very generous top limits  for pop and EIR